[quant][graphmode] Use quantized::batch_norm in graph mode

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40038 [quant][graphmode] Support another use pattern of mean
* #39919 [quant] Support general op modules with inplace options
* #39925 [quant][graphmode] Support quantizing `repeat`
* #39924 [quant][graphmode] Support squeeze/unsqueeze
* #39923 [quant][graphmode] Run RemoveRedundantDequantize in the end
* #39915 [quant][graphmode] Pass debug option into insert_quant_dequant pass
* **#39911 [quant][graphmode] Use quantized::batch_norm in graph mode**
* #39910 [quant] add quantized::batch_norm
* #39895 [quant][graphmode] Support prim::TupleUnpack and prim::TupleConstruct

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

jerryzh168 committed:
[quant][graphmode] Use quantized::batch_norm in graph mode

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]

vkuzo commented on torch/csrc/jit/passes/quantization/insert_observers.cpp:
```diff
@@ -718,8 +750,10 @@ graph(%self, %a, %b):
           inplace_add_aten_relu, inplace_add_aten_relu_,
-          nn_bn_nn_relu,         nn_bn_f_relu,
-          nn_bn_aten_relu,       nn_bn_aten_relu_,
+          nn_bn2d_nn_relu,       nn_bn2d_f_relu,
```
are we adding bn1d later?

jerryzh168 replied:
if this is used by user then we should implement them. right now we don't have plans to do that

jerryzh168 replied:
right now quantized::batch_norm1d is not implemented

jerryzh168 committed:
Update on "[quant][graphmode] Use quantized::batch_norm in graph mode"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22012282](https://our.internmc.facebook.com/intern/diff/D22012282)

[ghstack-poisoned]
